framework/src/org/apache/cordova/CordovaInterfaceImpl.java

```diff
@@ -115,6 +115,10 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent intent)
             savedResult = new ActivityResultHolder(requestCode, resultCode, intent);
             if (pluginManager != null) {
                 callback = pluginManager.getPlugin(initCallbackService);
+                if(callback != null) {
+                    callback.onRestoreStateForActivityResult(savedPluginState.getBundle(callback.getServiceName()),
```

Comment:
I believe so. This will only be called after the Activity was destroyed in which case `onSaveInstanceState()` will have been called and the `savedPluginState` bundle saved. I believe Activity results are only delivered after `onCreate()` is called, so the Bundle will have already been retrieved (but I can double check that). It is the responsibility of the plugin to check if the bundle they receive is null.